Framework/Source/Sample.h

```diff
@@ -70,146 +72,79 @@ namespace Falcor
+        /* Callback inherited from SampleCallbacks                                 */
+        /************************************************************************/
+        RenderContext::SharedPtr getRenderContext() override { return mpRenderContext; }
+        Fbo::SharedPtr getCurrentFbo() override { return mpTargetFBO; }
+        Gui* getGui() override { return mpGui.get(); }
+        float getCurrentTime() override { return mCurrentTime; }
+        void resizeSwapChain(uint32_t width, uint32_t height) override;
+        bool isKeyPressed(const KeyboardEvent::Key& key) override;
+        float getFrameRate() override { return mFrameRate.getAverageFrameTime(); }
+        float getLastFrameTime() override { return mFrameRate.getLastFrameTime();  }
+        uint64_t getFrameID() override { return mFrameRate.getFrameCount(); }
+        void renderText(const std::string& str, const glm::vec2& position, glm::vec2 shadowOffset = vec2(1)) override;
+        std::string getFpsMsg() override;
+        void toggleText(bool showText) override { mShowText = showText && gpDevice; }
+        void toggleUI(bool showUI) override { mShowUI = showUI && gpDevice; }
+        void setDefaultGuiSize(uint32_t width, uint32_t height) override;
+        void setCurrentTime(float time) override { mCurrentTime = time; }
+        ArgList getArgList() override { return mArgList; }
+        void setFixedTimeDelta(float newFixedTimeDelta) override { mFixedTimeDelta = newFixedTimeDelta; }
+        float getFixedTimeDelta() override  { return mFixedTimeDelta; }
+        std::string captureScreen(const std::string explicitFilename = "", const std::string explicitOutputDirectory = "") override;
+        void shutdown() override { if (mpWindow) { mpWindow->shutdown(); } }
+        
+        //Any cleanup required by renderer if its being shut down early via testing 
+        void onTestShutdown() override { mpRenderer->onTestShutdown(mpSampleTest.get()); }
 
-        /** Show/hide the UI
-        */
-        void toggleUI(bool showUI) { mShowUI = showUI && gpDevice; }
+        //Non inherited testing functions 
+        bool initializeTesting();
+        void beginTestFrame();
+        void endTestFrame();
 
-        /** Set the main GUI window size
+        /** Internal data structures
         */
-        void setSampleGuiWindowSize(uint32_t width, uint32_t height);
-
         Gui::UniquePtr mpGui;                               ///< Main sample GUI
         RenderContext::SharedPtr mpRenderContext;           ///< The rendering context
-        GraphicsState::SharedPtr mpDefaultPipelineState;    ///< The default pipeline state
-        Fbo::SharedPtr mpDefaultFBO;                        ///< The default FBO object
+        GraphicsState::SharedPtr mpDefaultPipelineState;    ///< The default pipeline 
+        Fbo::SharedPtr mpTargetFBO;                         ///< An FBO available to renderers
```

Comment:
This comment can be worded better. "An" implies there are multiple FBO's exposed